core/tee/tee_svc_storage.c

```diff
@@ -880,24 +892,29 @@ TEE_Result tee_svc_storage_reset_enum(uint32_t obj_enum)
 
 	res = tee_ta_get_current_session(&sess);
 	if (res != TEE_SUCCESS)
-		return res;
+		goto exit;
 
-	if (obj_enum == TEE_HANDLE_NULL)
-		return TEE_SUCCESS;
+	if (obj_enum == TEE_HANDLE_NULL) {
+		res = TEE_SUCCESS;
+		goto exit;
+	}
 
 	res = tee_svc_storage_get_enum(sess->ctx, obj_enum, &e);
 	if (res != TEE_SUCCESS)
-		return res;
+		goto exit;
 
 	res = tee_file_ops.closedir(e->dir);
 	e->dir = NULL;
-	if (res != 0)
-		return TEE_ERROR_GENERIC;
+	if (res) {
+		res = TEE_ERROR_GENERIC;
+		goto exit;
+	}
 
-	return TEE_SUCCESS;
+exit:
+	return res;
 }
 
-static TEE_Result tee_svc_storage_set_enum(char *d_name, struct tee_obj *o)
+static TEE_Result tee_svc_storage_set_enum(struct tee_obj *o, char *d_name)
```

Comment:
The `tee_svc_` prefix should be reserved for the syscall functions listed in `core/arch/arm/tee/arch_svc.c`, just drop it here and below.